crates/solvers/src/boundary/naive.rs

```diff
@@ -0,0 +1,239 @@
+    model::order::{
+        LimitOrderClass,
+        Order,
+        OrderClass,
+        OrderData,
+        OrderKind,
+        OrderMetadata,
+        OrderUid,
+    },
+    num::{BigRational, One},
+    solver::{
+        liquidity::{
+            slippage::{SlippageCalculator, SlippageContext},
+            AmmOrderExecution,
+            ConstantProductOrder,
+            Exchange,
+            LimitOrder,
+            LimitOrderId,
+            LiquidityOrderId,
+            SettlementHandling,
+        },
+        settlement::{external_prices::ExternalPrices, SettlementEncoder},
+        solver::naive_solver::multi_order_solver,
+    },
+    std::sync::{Arc, Mutex},
+};
+
+pub fn solve(
+    orders: &[&order::Order],
+    liquidity: &liquidity::Liquidity,
+) -> Option<solution::Solution> {
+    let pool = match &liquidity.state {
+        liquidity::State::ConstantProduct(pool) => pool,
+        _ => return None,
+    };
+
+    let boundary_orders = orders
+        .iter()
+        .map(|order| LimitOrder {
+            id: match order.class {
+                order::Class::Market => LimitOrderId::Market(OrderUid(order.uid.0)),
+                order::Class::Limit => LimitOrderId::Limit(OrderUid(order.uid.0)),
+                order::Class::Liquidity => {
+                    LimitOrderId::Liquidity(LiquidityOrderId::Protocol(OrderUid(order.uid.0)))
+                }
+            },
+            sell_token: order.sell.token.0,
+            buy_token: order.buy.token.0,
+            sell_amount: order.sell.amount,
+            buy_amount: order.buy.amount,
+            kind: match order.side {
+                order::Side::Buy => OrderKind::Buy,
+                order::Side::Sell => OrderKind::Sell,
+            },
+            partially_fillable: order.partially_fillable,
+            solver_fee: order.fee().amount,
+            settlement_handling: Arc::new(OrderHandler {
+                order: Order {
+                    metadata: OrderMetadata {
+                        uid: OrderUid(order.uid.0),
+                        class: match order.class {
+                            order::Class::Market => OrderClass::Market,
+                            order::Class::Limit => OrderClass::Limit(LimitOrderClass {
+                                surplus_fee: Some(order.fee().amount),
+                                ..Default::default()
+                            }),
+                            order::Class::Liquidity => OrderClass::Liquidity,
+                        },
+                        solver_fee: order.fee().amount,
+                        ..Default::default()
+                    },
+                    data: OrderData {
+                        sell_token: order.sell.token.0,
+                        buy_token: order.buy.token.0,
+                        sell_amount: order.sell.amount,
+                        buy_amount: order.buy.amount,
+                        fee_amount: order.fee().amount,
```

Comment:
`order.fee().amount` is used to initialize the user signed `fee_amount` as well as the `solver_fee` that can be scaled by the `autopilot` to promote CoWs. Since those values don't have to be identical it looks like an error.

Reply:
Added comment (see thread 👇).